app/controllers/fosters_controller.rb

```diff
@@ -48,7 +48,11 @@ def update
 
   # DELETE /fosters/1 or /fosters/1.json
   def destroy
-    @foster.destroy
+    Foster.transaction do
+      Home.find_by(foster_id: params[:id]).destroy
```

Comment:
I think the process should be different somehow but I'm not sure what yet.
Our model has a foster tied to a home, but that same home could be tied to other fosters. For example in the case of a shelter or a family who shares a home.
We would want to delete the home only if no other fosters were tied to it.
I think we want to allow many to many for homes and fosters.

Reply:
I can definitely see many fosters for one home based on the examples you mentioned. I was thinking for a bit what one foster having many homes would look like.

I recall us discussing someone having a summer home or something like that, but may not have been a common use case. However, I can see the possibility of a volunteer fostering a dog in a shelter and also fostering dogs at their home. I think those two examples validate modeling many homes for one foster.

From an implementation perspective, I can add the `:has_many` attribute to both tables. I imagine a join table will have to be created as well. I'm not sure if I have to do that myself or if rails will take care of that, but I'll figure it out.

Does that sound like a good direction to go in?

Reply:
Yeah it's that second scenario I was wondering about. Fostering a dog at home and being in charge of  one at a shelter.

I think we might want a join model here:
FosterHome belongs_to :foster
FosterHome belongs_to :home
Foster has_many :foster_homes
Foster has_many :homes, through: :foster_homes
Home has_many :foster_homes
Home has_many :fosters, through: :foster_homes

Reply:
I have implemented this relationship with my latest push